### DISCUSSION TITLE
enhancement request : show all consult-line matches in the buffer

### DISCUSSION THREAD

**Discussion by rileyrg** [minad/consult#905] [Ideas]
isearch highlights all matches in the buffer you're searching.

If consult-line could do that it would certainly replace isearch for me. So as I scrroll down my consult line candidates the buffer shows all matches in that locality.

![screenshot-latest](https://github.com/minad/consult/assets/362891/2ac3aedd-7d4b-4794-b4f5-f70dfdcc70a9)

**Comment by minad** [comment]
Highlighting all matches has been requested before and there has been a patch in #745. The difficulty is to implement this efficiently.

**Reply by rileyrg** [reply]
OK. Great library.

**Comment by hab25** [comment]
While this enhancement doesn't yet exist, one can workaround by executing
```emacs-lisp
(consult-line-multi
  (list
  :predicate (apply-partially #'eq (current-buffer))))
```

Caveats:
- As per @minad's comment performance might be unnacceptably worse than `consult-line` for some cases.

- When using this, prompt annoyingly shows "Go to line (1 buffer):" (would be better for prompt string to be equal to `consult-line`'s "Go to line:", since in this case there should be no gain of information from the display of " (1 buffer)".
